apps/coscrad-frontend/src/components/resources/factories/thumbnail-resource-detail-presenter-factory.tsx

```diff
@@ -11,31 +12,34 @@ import { TranscribedAudioDetailThumbnailPresenter } from '../transcribed-audio/t
 import { VocabularyListDetailThumbnailPresenter } from '../vocabulary-lists/vocabulary-list-detail.thumbnail.presenter';
 
 // TODO Define thumbnail specific presenters
-const lookupTable: { [K in ResourceType]: FunctionalComponent } = {
-    [ResourceType.bibliographicReference]: BibliographicReferenceDetailThumbnailPresenter,
-    [ResourceType.book]: BookDetailThumbnailPresenter,
-    [ResourceType.mediaItem]: MediaItemDetailPresenter,
-    [ResourceType.photograph]: PhotographDetailThumbnailPresenter,
-    [ResourceType.song]: SongDetailPresenter,
-    [ResourceType.spatialFeature]: SpatialFeatureDetailPresenter,
-    [ResourceType.term]: TermDetailThumbnailPresenter,
-    [ResourceType.transcribedAudio]: TranscribedAudioDetailThumbnailPresenter,
-    [ResourceType.vocabularyList]: VocabularyListDetailThumbnailPresenter,
+const lookupTable: { [K in CategorizableType]: FunctionalComponent } = {
+    [CategorizableType.bibliographicReference]: BibliographicReferenceDetailThumbnailPresenter,
```

Comment:
I wonder if we could use a tiny bit of OOP \ polymorphism to avoid these lookup tables all over the place? Maybe this would be our top level abstract factory that can
- build a thumnail view
- build a full-view
???